lib/watir/elements/element.rb

```diff
@@ -406,7 +406,11 @@ def visible?
       @element.displayed?
     rescue Selenium::WebDriver::Error::StaleElementReferenceError
       reset!
+      Watir.logger.deprecate "Getting an exception from #visible? when an Element does not exist", "#present?"
       raise unknown_exception
+    rescue UnknownObjectException, UnknownFrameException
+      Watir.logger.deprecate "Getting an exception from #visible? when an Element does not exist", "#present?"
```

Comment:
If users just switch `#visible?` to `#present?`, won't that be a change in behaviour? Shouldn't we provide a solution that allows users to maintain their current behaviour - eg `e.exists? ? e.visible? : raise(Watir::Exception::UnknownObjectException, "unable to locate element: #{e.inspect}")`

Reply:
Right, but one of the reasons for changing it is to change the behavior away from something that likely isn't actually desired, so I don't want them to just blindly replace `#visible?` with that ternary.

The other big concern I have with this approach is that some users might have tests that fail when an exception is rescued. So a user can run this code, all the tests pass and they don't get a deprecation notice. Then we update to alias `#visible?` with `#present?` and the System breaks, but the tests keep passing because it no longer throws the exception.

Really the only way to actually avoid that is to deprecate `#visible?`. I really want to deprecate it. Or at the very least throw a warning that "You are using `#visible?` when you probably want to use `#present?`". Maybe we can add a feature to the logger to make it so we can turn off specific warnings for people to silence specific warnings.

We also need to see how this affects PO gem before merging.

Reply:
I agree that deprecation would be easier (I had voted such last year). However, I think there are enough code bases and examples that warrants keeping `#visible?`. I'm guessing few of those usages are relying on raising the exception. 

Given it's hard to predict how people are using the exception, what if we just warn users that the behaviour is going to change? Don't provide a solution. Alternatively, provide a wiki or blog post link, which provides a longer discussion/examples.

Reply:
Yeah, I think a general warning with a blog post link is a good idea.